src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.Collections.cs

```diff
@@ -57,7 +60,7 @@ public void GetListNullValues()
             Assert.Empty(list);
         }
 
-        [Fact]
+        [ConditionalFact(typeof(TestHelpers), nameof(TestHelpers.NotSourceGenMode))] // Ensure exception messages are in sync
```

Comment:
>// Ensure exception messages are in sync [](http://example.com/codeflow?start=85&length=40)

what this means? if the difference only in the message, then you may enable the test in all cases and track fixing the messages.